### ISSUE TITLE
[Question] Coc-go vs go lsp settings

### ISSUE THREAD

**Issue by ivanddm** [josa42/coc-go#153]
Hi,

I just started developing in Go and I'm trying to setup my environment in vim/nvim.

I'm trying this extension and although it installs properly and loads `gopls`, it ignores the basic settings suggested in the readme page. Basically, with the following settings, errors are linted and completions are triggered but only if I manually add the package import statement first: 

```
"go.goplsOptions": {
  "completeUnimported": true,
  "gofumpt": true,
  "usePlaceholders": true,
},
```

With `"completeUnimported": true,` shouldn't completion work without importing the modules first? Also, formatting doesn't work, at least not out of the box.

However, without this extension and the following setting in `coc-settings.json`, both formatting and unimported packages completion is working just fine:

```
"coc.preferences.formatOnSaveFiletypes": [ "go" ],
"languageserver": {
  "go": {
    "command": "gopls",
    "rootPatterns": ["go.mod"],
    "filetypes": ["go"],
  }
},
```

I'm just wondering what setting should I add to get the `coc-go` extension working at least with these two features (formatting, and completion of unimported packages).

Otherwise, and sorry for asking probably a basic question here, but would like to know what features the `coc-go` extension adds to the vanilla go coc-nvim language server configuration.

Thanks in advance!

**Comment by luong-komorebi** [comment]
The general advice is when you are using coc-go, do not config in `coc-settings.json` with the instruction from `gopls` instruction page. 

Coc-go is born to do all the heavy lifting configuration so that coc can talks to gopls. The instruction on the `gopls` page is for people who did not know about coc-go

**Comment by ivanddm** [comment]
Yes, that's what I did, while using Coc-go I remove the config in `coc-settings.json`. So basically, with just the extension and no configuration in `coc-settings.json` whatsoever the extension is not working for me as described above in terms of completion of unimported packages.

**Comment by luong-komorebi** [comment]
Can you share your current version of go and gopls ? 
Try to run some commands like `:CocCommand go.version` to get the information

**Comment by ivanddm** [comment]
I have the latest versions, I think (coc-go 0.13.3, gopls 0.7.0). But actually, I just installed the extension again, and it seems to be working now. I think the issue was that when I first installed the extension, I didn't have any modules installed in my gopath, such as goimports, etc... Later on, I installed `vim-go`  instead, which automatically downloaded and added those packages in my gopath. So now, after disabling `vim-go` and installing again `coc-go`, the latter seems to be working as expected.

Thanks in any case. I'm closing this issue for now.
